### PR TITLE
feat: Add users to waitlist

### PR DESCRIPTION
**Pull request by EAZYLINK**
# What does the PR do?
This PR is concerned with adding users to waitlist

## How should this endpoint be tested?
Send a POST request to /api/v1/waitlist with the request body as follows:
```
{
"email": "string",
"fullName": "string"
}
```
## Checklist of what I did

- [x] Implemented the endpoint to join a waitlist
- [x] Validated the request body and returned validation errors if anyone fails
- [x] Prevented duplicate entry by making the email unique
- [x] Prevented empty entry for full name by adding non-empty decorator
- [x] Stored the user entry in the database
- [x] Implemented a send mail logic to send confirmation mail to registered users
- [x] Returned a success response for successful operation
- [x] Added unit test for waitlist service
## Reference Issue
https://github.com/hngprojects/hng_boilerplate_nestjs/issues/106

### CODE REVIEW

**Comment by davidolanrewaju** [comment]
Fix your workflow

File: src/modules/waitlist/dto/create-waitlist-user.dto.ts
Comment (davidolanrewaju) [inline src/modules/waitlist/dto/create-waitlist-user.dto.ts:1]: Include documentation on your dtos

Comment (EAZYLINK) [reply src/modules/waitlist/dto/create-waitlist-user.dto.ts:1]: I don't understand, what kind of documentation?

File: db/migrations/1721832121420-migration.ts
Comment (Heba-WebDev) [inline db/migrations/1721832121420-migration.ts:1]: Migration files are no longer allowed so please delete this.

File: db/migrations/1721832141457-migration.ts
Comment (Heba-WebDev) [inline db/migrations/1721832141457-migration.ts:1]: This as well, please delete it

File: db/migrations/1721832121420-migration.ts
Comment (EAZYLINK) [reply db/migrations/1721832121420-migration.ts:1]: done

File: db/migrations/1721832141457-migration.ts
Comment (EAZYLINK) [reply db/migrations/1721832141457-migration.ts:1]: done

**Comment by CFCIfe** [comment]
@EAZYLINK you marked the comments as resolved but have not pushed the changes. Its not reflecting here

**Review by Heba-WebDev — DISMISSED** [review]

**Review by Heba-WebDev — DISMISSED** [review]

**Review by Heba-WebDev — DISMISSED** [review]

**Review by Heba-WebDev — APPROVED** [review]

**Review by Cobidex — COMMENTED** [review]
Please update the env.example file with the CLIENT_URL environment variable you are introducing in this PR

**Pull request closed without merging**